lib/grpcServer/handlers/tx-filter-stream/subscribeToTransactionsWithProofsHandlerFactory.js

```diff
@@ -164,17 +155,62 @@ function subscribeToTransactionsWithProofsHandlerFactory(
       );
     }
 
+    // If block height is specified instead of block hash, we obtain block hash by block height
+    if (fromBlockHash === '') {
+      if (fromBlockHeight === 0) {
+        throw new InvalidArgumentGrpcError('minimum value for `fromBlockHeight` is 1');
+      }
+
+      // we don't need to check bestBlockHeight because getBlockHash throws
+      // an error in case of wrong height
+      try {
+        fromBlockHash = await coreAPI.getBlockHash(fromBlockHeight);
+      } catch (e) {
+        if (e.code === -8) {
+          // Block height out of range
+          throw new InvalidArgumentGrpcError('fromBlockHeight is bigger than block count');
+        }
+
+        throw e;
+      }
+    }
+
+    // Send historical transactions
+    let fromBlock;
+
+    try {
+      fromBlock = await coreAPI.getBlock(fromBlockHash);
```

Comment:
We also need to check that fromBlock.confirmations !== -1 otherwise we need to throw an error.